esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -898,6 +898,28 @@ void WsWuInfo::getGraphTimingData(IArrayOf<IConstECLTimingData> &timingData)
         it->playProperties(visitor);
 }
 
+void WsWuInfo::getServiceNames(IEspECLWorkunit &info, unsigned long flags)
+{
+    if (!(flags & WUINFO_IncludeServiceNames))
+        return;
+
+    StringArray serviceNames;
+    WuScopeFilter filter;
+    filter.addOutputAttribute(WaServiceName);
+    filter.finishedFilter();
+    Owned<IConstWUScopeIterator> it = &cw->getScopeIterator(filter);
+    ForEach(*it)
+    {
+        if (it->getScopeType() != SSTactivity)
```

Comment:
This filter should be added to the WuScopeFilter, rather than being checked in the iterator loop:

    filter.addScopeType("activity");

I have also merged the PR that allows a restriction on the attribute, so you can also add

    filter.addRequiredAttr(WaServiceName);